circleci: failed tests: TestRaftTransportCircuitBreaker

The following test appears to have failed:

[#21528](https://circleci.com/gh/cockroachdb/cockroach/21528):

```
I160818 18:13:49.458132 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:50270->127.0.0.1:50805: use of closed network connection
--- PASS: TestInOrderDelivery (0.53s)
=== RUN   TestRaftTransportCircuitBreaker
W160818 18:13:49.460971 gossip/gossip.go:1039  not connected to cluster; use --join to specify a connected node
W160818 18:13:49.461566 storage/raft_transport.go:256  raft transport stream to node 2 failed: unable to lookup descriptor for node 2
panic: test timed out after 2m0s

goroutine 20441 [running]:
panic(0x153d1e0, 0xc4225f41e0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:918 +0x10b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:154 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc42016e240, 0x16f0fc8, 0x1f, 0x18477a0, 0xc420271a01)
	/usr/local/go/src/testing/testing.go:647 +0x316
testing.RunTests.func1(0xc42016e240)
	/usr/local/go/src/testing/testing.go:793 +0x6d
testing.tRunner(0xc42016e240, 0xc420045bc0)
	/usr/local/go/src/testing/testing.go:610 +0x81
testing.RunTests(0x1848a40, 0x2051120, 0x136, 0x136, 0xc420276500)
	/usr/local/go/src/testing/testing.go:799 +0x2f5
testing.(*M).Run(0xc420045ee8, 0xc420271d30)
	/usr/local/go/src/testing/testing.go:743 +0x85
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc420045ee8)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:57 +0x1ff
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:682 +0xc6

goroutine 17 [syscall, 2 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2396d40)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1016 +0x77
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:599 +0x95

goroutine 20430 [chan receive]:
github.com/cockroachdb/cockroach/util/netutil.ListenAndServeGRPC.func1()
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:51 +0x6f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc421192900, 0xc422615620)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66

goroutine 20427 [chan receive]:
github.com/cockroachdb/cockroach/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:122 +0x66
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc421192900, 0xc421f5ad40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66

goroutine 20429 [IO wait]:
net.runtime_pollWait(0x7f020b3e54d8, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc421c56bc0, 0x72, 0xc42157c5f0, 0xc420012098)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc421c56bc0, 0x205c9a0, 0xc420012098)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).accept(0xc421c56b60, 0x0, 0x2059d20, 0xc4204beae0)
	/usr/local/go/src/net/fd_unix.go:419 +0x238
net.(*TCPListener).accept(0xc4227429f0, 0xc42157c6d8, 0xc42157c6e8, 0x614580)
	/usr/local/go/src/net/tcpsock_posix.go:132 +0x2e
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc421192900, 0xc4229aeb20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66

goroutine 20428 [chan receive]:
github.com/cockroachdb/cockroach/util/netutil.ListenAndServeGRPC.func1()
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:51 +0x6f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc421192900, 0xc4226152f0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66

goroutine 20431 [IO wait]:
net.runtime_pollWait(0x7f020b46d5c0, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc421c56ca0, 0x72, 0xc422b4b5f0, 0xc420012098)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc421c56ca0, 0x205c9a0, 0xc420012098)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).accept(0xc421c56c40, 0x0, 0x2059d20, 0xc4204beb00)
	/usr/local/go/src/net/fd_unix.go:419 +0x238
net.(*TCPListener).accept(0xc422742a00, 0xc422b4b6d8, 0xc422b4b6e8, 0x614580)
	/usr/local/go/src/net/tcpsock_posix.go:132 +0x2e
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc421192900, 0xc4229aebe0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66

goroutine 20426 [chan receive]:
github.com/cockroachdb/cockroach/storage_test.TestRaftTransportCircuitBreaker(0xc420840540)
	/go/src/github.com/cockroachdb/cockroach/storage/raft_transport_test.go:454 +0x523
testing.tRunner(0xc420840540, 0x18477a0)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
FAIL	github.com/cockroachdb/cockroach/storage	120.040s
=== RUN   TestBatchBasics
I160818 18:13:39.548125 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160818 18:13:39.549742 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchRepr
I160818 18:13:39.552654 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160818 18:13:39.553742 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.00s)
=== RUN   TestBatchGet
I160818 18:13:39.555769 storage/engine/rocksdb.go:354  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/raft_transport_test.go
+++ b/storage/raft_transport_test.go
@@ -413,16 +413,10 @@ func TestRaftTransportCircuitBreaker(t *testing.T) {
 	}
 	clientTransport := rttc.AddNode(clientReplica.NodeID)
 
-	// The transport is set up asynchronously, so we expect Send to return true here.
+	// The transport is set up asynchronously, so we expect the first
+	// Send to return true here.
 	if !rttc.Send(clientReplica, serverReplica, 1, raftpb.Message{Commit: 1}) {
 		t.Errorf("unexpectedly failed sending while connection is being asynchronously established")
-	}
-
-	// None should go through as the receiving node's address has not been gossiped.
-	select {
-	case req := <-serverChannel.ch:
-		t.Fatalf("should not have received any Raft messages from client: %s", req)
-	default:
 	}
 
 	// However, sending repeated messages should begin dropping once
@@ -437,24 +431,22 @@ func TestRaftTransportCircuitBreaker(t *testing.T) {
 	// Now, gossip address of server.
 	rttc.GossipNode(serverReplica.NodeID, serverAddr)
 
-	// When the circuit breaker tripped, the queue would have been deleted.
-	// So gossiping the server address won't magically make the request appear.
-	select {
-	case req := <-serverChannel.ch:
-		t.Fatalf("should not have received any Raft messages from client: %s", req)
-	default:
-	}
-
-	// Reset the circuit breaker & resend and verify message arrives at server.
-	clientTransport.GetCircuitBreaker(serverReplica.NodeID).Reset()
-	if !rttc.Send(clientReplica, serverReplica, 1, raftpb.Message{Commit: 2}) {
-		t.Errorf("sent raft message was unexpectedly dropped")
-	}
-
-	req := <-serverChannel.ch
-	if req.Message.Commit != 2 {
-		t.Errorf("expected message 2; got %s", req)
-	}
+	// Keep sending commit=2 until breaker resets and we receive the
+	// first instance. It's possible an earlier message for commit=1
+	// snuck in.
+	util.SucceedsSoon(t, func() error {
+		if !rttc.Send(clientReplica, serverReplica, 1, raftpb.Message{Commit: 2}) {
+			clientTransport.GetCircuitBreaker(serverReplica.NodeID).Reset()
+		}
+		select {
+		case req := <-serverChannel.ch:
+			if req.Message.Commit == 2 {
+				return nil
+			}
+		default:
+		}
+		return errors.Errorf("expected message commit=2")
+	})
 }
 
 // TestRaftTransportIndependentRanges ensures that errors from one

